handsontable/src/plugins/emptyDataState/__tests__/options/message.spec.js

```diff
@@ -0,0 +1,465 @@
+    handsontable({
+      data: [],
+      emptyDataState: {
+        message: 'Custom empty state message',
+      },
+    });
+
+    const emptyDataStatePlugin = getPlugin('emptyDataState');
+
+    expect(emptyDataStatePlugin.isVisible()).toBe(true);
+    expect(getEmptyDataStateContainerElement().querySelector('.ht-empty-data-state__title').textContent)
+      .toBe('Custom empty state message');
+  });
+
+  it('should show custom object message with title and description', async() => {
+    handsontable({
+      data: [],
+      emptyDataState: {
+        message: {
+          title: 'Custom Title',
+          description: 'Custom description text',
+        },
+      },
+      filters: true,
+    });
+
+    const filtersPlugin = getPlugin('filters');
+    const emptyDataStatePlugin = getPlugin('emptyDataState');
+
+    expect(emptyDataStatePlugin.isVisible()).toBe(true);
+    expect(getEmptyDataStateContainerElement().querySelector('.ht-empty-data-state__title').textContent)
+      .toBe('Custom Title');
+    expect(getEmptyDataStateContainerElement().querySelector('.ht-empty-data-state__description').textContent)
+      .toBe('Custom description text');
+
+    // Add data back and test filtered state
+    hot().updateSettings({
+      data: createSpreadsheetData(5, 5),
+    });
+
+    // Apply filter that hides all data
+    filtersPlugin.addCondition(0, 'eq', ['nonexistent']);
+    filtersPlugin.filter();
+
+    expect(emptyDataStatePlugin.isVisible()).toBe(true);
+    expect(getEmptyDataStateContainerElement().querySelector('.ht-empty-data-state__title').textContent)
+      .toBe('Custom Title');
+    expect(getEmptyDataStateContainerElement().querySelector('.ht-empty-data-state__description').textContent)
+      .toBe('Custom description text');
+  });
+
+  it('should show custom object message with actions', async() => {
+    const callbackSpy = jasmine.createSpy('callback');
+
+    handsontable({
+      data: [],
+      emptyDataState: {
+        message: {
+          title: 'No Data',
+          description: 'Please add some data',
+          actions: [{
+            text: 'Add Data',
+            type: 'primary',
+            callback: callbackSpy,
+          }],
+        },
+      },
+    });
+
+    const emptyDataStatePlugin = getPlugin('emptyDataState');
+
+    expect(emptyDataStatePlugin.isVisible()).toBe(true);
+    expect(getEmptyDataStateContainerElement().querySelector('.ht-empty-data-state__title').textContent)
+      .toBe('No Data');
+    expect(getEmptyDataStateContainerElement().querySelector('.ht-empty-data-state__description').textContent)
+      .toBe('Please add some data');
+
+    const actionButton = getEmptyDataStateContainerElement().querySelector('.ht-button--primary');
+
+    expect(actionButton).toBeDefined();
+    expect(actionButton.textContent).toBe('Add Data');
+    expect(actionButton).toHaveClass('ht-button', 'ht-button--primary');
+
+    actionButton.click();
```

Comment:
There is a `simulateClick` helper available, which is better as it simulates all click-related events (connected with `mouse` events) in proper order to simulate a real click.
```suggestion
    await simulateClick(actionButton);
```